File: app/src/main/java/art/coded/wireframe/ui/work/WorkUtilities.java
package art.coded.wireframe.ui.work;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.os.Build;

final class WorkUtilities {

    static void loadNotificationChannel(
            Context context,
            String channelId,
            String channelName,
            String channelDescription,
            int importance) {

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {

            NotificationChannel notificationChannel =
                    new NotificationChannel(
                            channelId, channelName, importance);
            notificationChannel.setDescription(channelDescription);

            NotificationManager notificationManager =
                    (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);

            if (notificationManager != null) {
                notificationManager.createNotificationChannel(notificationChannel);
            }
        }
    }
}

File: app/src/ktx/java/art/coded/wireframe/view/work/WorkUtilities.kt
package art.coded.wireframe.view.work

import android.os.Build
import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context

internal object WorkUtilities {
    fun loadNotificationChannel(
        context: Context,
        channelId: String?,
        channelName: String?,
        channelDescription: String?,
        importance: Int
    ) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val notificationChannel = NotificationChannel(
                channelId, channelName, importance
            )
            notificationChannel.description = channelDescription
            val notificationManager =
                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager?.createNotificationChannel(notificationChannel)
        }
    }
}